apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/grpc/ToProtoValueCodecs.java

```diff
@@ -611,6 +611,11 @@ public QueryOuterClass.Value protoValueFromStrictlyTyped(Object value) {
       if (value instanceof String) {
         return protoValueFromStringified((String) value);
       }
+      // Should only accept integral numbers; Jackson will expose these as Integers and Longs
+      // (unfortunately "java.lang.Number" has no convenience methods for determining this)
+      if (value instanceof Long || value instanceof Integer) {
+        return Values.of(((Number) value).longValue());
+      }
```

Comment:
how can a timestamp be represented in an Integer? The max value is not enough.

Reply:
If caller specifies a number that fits in `int` it gets bound to `Integer`, that's the only reason.
In most cases it's big enough to become `Long`.
But such small values are technically valid (from early 1970s) so should, I think, be accepted.

Reply:
> OK, was not aware that early 1970s can fit into an int..

Yeah it's just because the textual representation (like `0` for 1970-01-01 UTC) fits... JSON has not types for different classes. So parser just sees a "small enough" number, without context. Coercion to `long` is done if and as needed for properties, but when read as `java.lang.Object` it'll choose smaller choice (does not try to use `Short` etc tho, just `int` and `long` and `BigInteger` for beyond).